Makefile

```
@@ -15,3 +15,7 @@ fmt:
 
 clean:
 	rm rate-limiting.wasm
+
+test: rate-limiting.wasm
+	$(GO) test -v ./...
```

```suggestion
	$(GO) test -race -v ./...
```
Better to add -race from the start than having a panic in production once it reaches a certain threshold for the race condition to appear.
Altho it should be noted that if the binary itself (for integration testing) isn't compiled with -race but launched separatly, it won't detect a race condition I think. Might want to add a small unit test that would used 2 rate limit ctx for example to be sure.
I mention race condition because I see that you return ctx.limit and also modify this (but I'm on my phone so hard time seing if there is a race condition, probably not).